### PR TITLE
[ENG-1354, 1415] Fixes bottom sidesheet and ReactFlow alignment issues

### PR DESCRIPTION
**Pull request by vsreekanti**
## Describe your changes and why you are making these changes

This PR fixes two bugs around the layout of the ReactFlow canvas and the bottom sidesheet:
* The fix for ENG-1354 is around calling `fitView` at the right time. Because we were previously calling `fitView` on `componentDidMount`, it didn't actually work on the first load because the nodes and edges were loaded in _after_ the component was mounted. This PR changes the `fitView` effect so that it also is called when the node/edge positions are changed, which causes the view to be fit correctly (after a small lag) on first load.
* This PR also fixes ENG-1415 simplifies absolute positioning of the bottom side sheet so that it properly aligns with the ReactFlow viewport (see screenshot below): 

(Ignore the miscolored artifacts; that's because my local backend isn't updated.)
image

## Checklist before requesting a review
- [x] I have created a descriptive PR title. The PR title should complete the sentence "This PR...".
- [x] I have performed a self-review of my code.
- [x] I have included a small demo of the changes. For the UI, this would be a screenshot or a Loom video.
- [N/A] If this is a new feature, I have added unit tests and integration tests.
- [N/A] I have run the integration tests locally and they are passing.
- [x] I have run the linter script locally (See `python3 scripts/run_linters.py -h` for usage).
- [x] All features on the UI continue to work correctly.
- [x] Added one of the following CI labels:
    - `run_integration_test`: Runs integration tests
    - `skip_integration_test`: Skips integration tests (Should be used when changes are ONLY documentation/UI)

### CODE REVIEW

File: src/ui/common/src/components/workflows/ReactFlowCanvas.tsx [hunk 1]
```diff
@@ -19,23 +18,23 @@ type ReactFlowCanvasProps = {
...
+  );
+
+  const { fitView, viewportInitialized } = useReactFlow();
   useEffect(() => {
     fitView();
-  }, []);
+  }, [dagPositionState]);
 
   useEffect(() => {
     // NOTE(vikram): There's a timeout here because there seems to be a
```
Comment (agiron123) [inline src/ui/common/src/components/workflows/ReactFlowCanvas.tsx:40]: timeout still needed?

File: src/ui/common/src/components/layouts/sidebar/AqueductSidebar.tsx [hunk 2]
```diff
@@ -151,9 +151,8 @@ export const AqueductSidebar: React.FC<Props> = ({
     // open bottom to top
     bottomAligned: {
       position: 'absolute',
-      right: bottomSideSheetOffset,
+      left: MenuSidebarOffset,
```
Comment (agiron123) [inline src/ui/common/src/components/layouts/sidebar/AqueductSidebar.tsx:154]: Not super urgent, but we'll need to add this to the EnterpriseMenuSidebar as well. we use different components on enterprise and OSS

**Review by agiron123 — APPROVED** [review]

**Comment by agiron123** [comment]
Looks good, remember to also update the version number of @aqueducthq/common here as well. 

We can say something like ```v0.0.11-rc.0``` so as not to have too many version numbers.

Here's how the folks at emotion do it:
image

File: src/ui/common/src/components/workflows/ReactFlowCanvas.tsx (hunk 1 above)
Comment (vsreekanti) [reply src/ui/common/src/components/workflows/ReactFlowCanvas.tsx:40]: I think this is still necessary. This has to do with a race condition between the CSS transition for the `div` and when the `fitView` call is made.

**Pull request merged**